lib/views/commit-view.js

```diff
@@ -115,6 +122,11 @@ export default class CommitView extends React.Component {
 
     const showAbortMergeButton = this.props.isMerging || null;
 
+    const modKey = process.platform === 'darwin' ? 'Cmd' : 'Ctrl';
+
+    // TODO: [ku 4/9/2018] fix tooltip target method (we're using `document.querySelector` because
+    // this.refCoAuthorToggle is temporarily null when the component is unmounted/remounted)
```

Comment:
It's a bit more complicated. That would result in no new tooltip once the DOM node was swapped out. I poked around a bit trying to figure out what's going on, but it was super late at night and my memory is fuzzy on it. I _think_ what's happening is that when we re-render the co-author icon DOM node is being replaced with a new DOM node. During this swap, `this.refCoAuthorToggle` is temporarily set to `null`. While this is the case, the Tooltip component tries to add the tooltip to the `null` target. So, rather than relying on the ref that gets nulled out, we use `document.querySelector('.github-CommitView-coAuthorToggle')` to get ahold of the target to add the tooltip to. 

Changing the `componentWillReceiveProps` to a `componentDidUpdate` seems to fix the issue and allow us to rely on the ref as opposed to querying the DOM. @smashwilson do you see any problems with this?

Reply:
> Changing the componentWillReceiveProps to a componentDidUpdate in the Tooltip component seems to fix the issue and allow us to rely on the ref as opposed to querying the DOM.

:sparkles: Perfect - especially because [`componentWillReceiveProps()` is being deprecated](https://reactjs.org/docs/react-component.html#unsafe_componentwillreceiveprops).